index.js

```diff
@@ -111,6 +111,33 @@ class ScmBase {
             process.nextTick(() => reject('not implemented'));
         });
     }
+
+    /**
+    * Fetch content of a file from an scm repo
+    * @method getFile
+    * @param  {Object}   config              Configuration
+    * @param  {String}   config.scmUrl       The scmUrl to get permissions on
+    * @param  {String}   config.path         The file in the repo to fetch
+    * @param  {String}   config.token        The token used to authenticate to the SCM
+    * @return {Promise}
+    */
+    getFile(config) {
+        const result = Joi.validate(config, dataSchema.plugins.scm.getFile);
+
+        if (result.error) {
+            return new Promise((resolve, reject) => {
+                process.nextTick(() => reject(result.error));
```

Comment:
A quick search online doesn't have anything bad. I read that promises are forced to run asynchronously